### ISSUE TITLE
Current schema selection fails for Snowflake connections

### ISSUE THREAD

**Issue by daniel-sali** [dbeaver/dbeaver#11450]
#### System information: 
- Operating system (distribution) and version: **Mac OS Mojave 10.14.6**
- DBeaver version  **Version 7.3.5.202102151746**
- Additional extensions **None**

#### Connection specification:
- Database name and version **Snowflake 3.12.9**
- Driver name **snowflake-jdbc-3.6.8.jar**
- Do you use tunnels or proxies (SSH, SOCKS, etc)? ** No **

#### Describe the problem you're observing:
If you open the current catalog/schema selection dialog to set/change the current working schema, the schema selection will fail with an error.

This is because selecting the schema gets translated to a `USE DATABASE ... ` statement instead of a `USE SCHEMA ... `
statement.

This is also the root cause of https://github.com/dbeaver/dbeaver/issues/11233

#### Steps to reproduce, if exist:
1. Establish a Snowflake connection.
2. Open a new SQL Script.
3. Open the "Choose catalog/schema" window by clicking on the current catalog selection button on the toolbar:
![Screenshot 2021-02-22 at 17 47 14](https://user-images.githubusercontent.com/13250673/108742097-a661a400-7537-11eb-89c4-7299a0af94ae.png)

4. Double-click on a schema (any), and you will get the following error:
![Screenshot 2021-02-22 at 17 47 29](https://user-images.githubusercontent.com/13250673/108742476-048e8700-7538-11eb-93ef-1e49b6044d47.png)

#### Include any warning/errors/backtraces from the logs

```log
2021-02-22 17:46:27.609 - Connect with 'SNOWFLAKE' (snowflake_jdbc-177ca3fde49-36b32575357541b0)
2021-02-22 17:46:27.610 - Execution context opened (SNOWFLAKE; Main; 5)
2021-02-22 17:46:28.054 - Execution context opened (SNOWFLAKE; Metadata; 6)
2021-02-22 17:46:28.399 - Error getting sqlStateType: null
2021-02-22 17:46:28.547 - Connected (snowflake_jdbc-177ca3fde49-36b32575357541b0, Snowflake [3.12.9])
2021-02-22 17:46:41.323 - Execution context opened (SNOWFLAKE; SQLEditor <Script.sql>; 7)
2021-02-22 17:46:50.978 - Error during job 'Change active database' execution
net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:
Object does not exist, or operation cannot be performed.
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:153)
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:77)
        at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:503)
        at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:380)
        at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:582)
        at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:266)
        at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:202)
        at net.snowflake.client.core.SFStatement.execute(SFStatement.java:877)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeInternal(SnowflakeStatementV1.java:331)
        at net.snowflake.client.jdbc.SnowflakePreparedStatementV1.execute(SnowflakePreparedStatementV1.java:535)
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.execute(JDBCPreparedStatementImpl.java:261)
        at org.jkiss.dbeaver.ext.generic.model.GenericExecutionContext.setDefaultSchema(GenericExecutionContext.java:315)
        at org.jkiss.dbeaver.ext.generic.model.GenericExecutionContext.setDefaultSchema(GenericExecutionContext.java:296)
        at org.jkiss.dbeaver.ext.generic.model.GenericExecutionContext.setDefaultSchema(GenericExecutionContext.java:1)
        at org.jkiss.dbeaver.model.exec.DBExecUtils.changeDefaultObject(DBExecUtils.java:396)
        at org.jkiss.dbeaver.model.exec.DBExecUtils.setExecutionContextDefaults(DBExecUtils.java:355)
        at org.jkiss.dbeaver.ui.actions.datasource.SelectActiveSchemaHandler$2.run(SelectActiveSchemaHandler.java:207)
        at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by uslss** [comment]
thanks for the report

**Comment by daniel-sali** [comment]
Additionally, the USE SCHEMA statements sent when opening a new Script for a schema with case-sensitive names are issued without quotes, resulting in an error:

```log
2021-02-25 20:23:07.421 - Error while initializing context state
org.jkiss.dbeaver.model.exec.DBCException: SQL Error [1003] [42000]: SQL compilation error:
syntax error line 1 at position 21 unexpected '/'.
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.useSchemaWithName(SnowflakeExecutionContext.java:51)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.useDefaultSchema(SnowflakeExecutionContext.java:38)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeDataSource.initializeContextState(SnowflakeDataSource.java:60)
        at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:148)
        at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.lambda$0(JDBCRemoteInstance.java:132)
        at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:168)
        at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.openIsolatedContext(JDBCRemoteInstance.java:130)
        at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$OpenContextJob.run(SQLEditor.java:619)
        at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:
syntax error line 1 at position 21 unexpected '/'.
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:124)
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:64)
        at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:434)
        at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:338)
        at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:482)
        at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:194)
        at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:131)
        at net.snowflake.client.core.SFStatement.execute(SFStatement.java:737)
        at net.snowflake.client.core.SFStatement.execute(SFStatement.java:635)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeUpdateInternal(SnowflakeStatementV1.java:189)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeLargeUpdate(SnowflakeStatementV1.java:169)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeUpdate(SnowflakeStatementV1.java:157)
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeUpdate(JDBCStatementImpl.java:358)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.useSchemaWithName(SnowflakeExecutionContext.java:48)
        ... 9 more

net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:
syntax error line 1 at position 21 unexpected '/'.
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:124)
        at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:64)
        at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:434)
        at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:338)
        at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:482)
        at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:194)
        at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:131)
        at net.snowflake.client.core.SFStatement.execute(SFStatement.java:737)
        at net.snowflake.client.core.SFStatement.execute(SFStatement.java:635)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeUpdateInternal(SnowflakeStatementV1.java:189)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeLargeUpdate(SnowflakeStatementV1.java:169)
        at net.snowflake.client.jdbc.SnowflakeStatementV1.executeUpdate(SnowflakeStatementV1.java:157)
        at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeUpdate(JDBCStatementImpl.java:358)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.useSchemaWithName(SnowflakeExecutionContext.java:48)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.useDefaultSchema(SnowflakeExecutionContext.java:38)
        at org.jkiss.dbeaver.ext.snowflake.model.SnowflakeExecutionContext.setDefaultCatalog(SnowflakeExecutionContext.java:29)
```

**Comment by flambchap** [comment]
Will be resolved in the scope of #11769.
